### PR TITLE
Gosec

### PR DESCRIPTION
**Pull request by wuurrd**
### What problem does this PR solve? 
#405 

### What is changed and how does it work?
Added a gosec target. Changed so Makefile test depends on it. It installs (if necessary) the gosec binary and runs it.

### Check List 
* [x] failpoint-enable before gosec
* [x] Fix actual gosec issues that are present
* [x] Add sanitizing on `exec.Command`

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### Request for comments:

```
Golang errors in file: [github.com/pingcap/chaos-mesh/pkg/time/time_linux.go]:

  > [line 96 : column 25] - Trace not declared by package ptrace



[github.com/pingcap/chaos-mesh/controllers/networkchaos/partition/types.go:18] - G505 (CWE-327): Blacklisted import crypto/sha1: weak cryptographic primitive (Confidence: HIGH, Severity: MEDIUM)
  > "crypto/sha1"


[github.com/pingcap/chaos-mesh/controllers/networkchaos/partition/types.go:241] - G401 (CWE-326): Use of weak cryptographic primitive (Confidence: HIGH, Severity: MEDIUM)
  > sha1.New()


[github.com/pingcap/chaos-mesh/pkg/webhook/config/config.go:171] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
  > os.Open(configFile)


[github.com/pingcap/chaos-mesh/pkg/chaosdaemon/util.go:183] - G204 (CWE-78): Subprocess launched with variable (Confidence: HIGH, Severity: MEDIUM)
  > exec.CommandContext(ctx, "nsenter", args...)


[github.com/pingcap/chaos-mesh/pkg/chaosdaemon/util.go:252] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
  > os.Open(statusPath)


[github.com/pingcap/chaos-mesh/pkg/pidfile/pidfile.go:44] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
  > ioutil.ReadFile(path)


[github.com/pingcap/chaos-mesh/controllers/networkchaos/partition/types.go:242] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > hasher.Write([]byte(nameRest))


[github.com/pingcap/chaos-mesh/pkg/chaosdaemon/util.go:265] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > fmt.Fscanf(reader, "%d %s %s %d", &pid, &comm, &state, &parent)


[github.com/pingcap/chaos-mesh/pkg/server/web.go:26] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > f.Close()


[github.com/pingcap/chaos-mesh/pkg/chaosdaemon/server.go:107] - G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)
  > srv.Shutdown(context.Background())


Summary:
   Files: 92
   Lines: 15077
   Nosec: 0
  Issues: 10
```

Which of these issues should be ignored - the errors unhandled are not critical? Are there enough tests that a test will break if we add sanitizing to exec, or is there a manual job to verify?

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
@wuurrd thank you for your contribution

**Comment by zhouqiang-cl** [comment]
@cwen0 I think we should first fix the CI. then merge this pr

File: Makefile [hunk 1]
```diff
@@ -109,6 +110,12 @@ manifests: controller-gen
 fmt: groupimports
 	$(CGOENV) go fmt ./...
 
+gosec:
+ifeq (,$(shell which gosec))
+	curl -sfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v2.2.0
```
Comment (yeya24) [inline Makefile:115]: Is it better to make this version a variable, not hard code it? WDYT

**Comment by wuurrd** [comment]
@zhouqiang-cl sure, but can you please advise what fixes you want to ignore vs. fix?

File: Makefile (hunk 1 above)
Comment (yeya24) [inline Makefile:115]: Hey, we manage all our external dependencies in a tools file, could you please follow this PR https://github.com/pingcap/chaos-mesh/pull/281 to update the Makefile? Thanks!

**Comment by yeya24** [comment]
> @zhouqiang-cl sure, but can you please advise what fixes you want to ignore vs. fix?

I think we can ignore this issues first. But then open another issue to track this security problems. WDYT? We need to make CI happy first...

**Comment by zhouqiang-cl** [comment]
> > @zhouqiang-cl sure, but can you please advise what fixes you want to ignore vs. fix?
> 
> I think we can ignore these issues first. But then open another issue to track these security problems, to see if these issues are really critical to us. WDYT? We need to make CI happy first...

OK. @cwen0 could you assigne someone first to fix the CI, then we will process this PR?

**Comment by zhouqiang-cl** [comment]
@wuurrd @rbtcollins thank you very much

**Comment by cwen0** [comment]
@wuurrd @rbtcollins Please fix conflicts

File: Makefile [hunk 2]
```diff
@@ -185,6 +188,8 @@ failpoint-ctl:
 	$(GO) get github.com/pingcap/failpoint/failpoint-ctl@v0.0.0-20200210140405-f8f9fb234798
 goimports:
 	$(GO) get golang.org/x/tools/cmd/goimports@v0.0.0-20200309202150-20ab64c0d93f
+gosec:
+	$(GO) get github.com/securego/gosec/cmd/gosec
```
Comment (Gallardot) [inline Makefile:192]: Do we need to specify a version for gosec? Like any other `$(GO) get`

Comment (yeya24) [reply Makefile:192]: Agree

**Pull request merged**